--- a/cmake/config-ix.cmake	2026-01-26 10:41:56.697112240 +0100
+++ b/cmake/config-ix.cmake	2026-01-26 10:43:04.427866274 +0100
@@ -66,16 +66,6 @@ if (C_SUPPORTS_NODEFAULTLIBS_FLAG)
                         moldname mingwex msvcrt)
     list(APPEND CMAKE_REQUIRED_LIBRARIES ${MINGW_LIBRARIES})
   endif()
-  if (NOT TARGET unwind)
-    # Don't check for a library named unwind, if there's a target with that name within
-    # the same build.
-    check_library_exists(unwind _Unwind_GetRegionStart "" COMPILER_RT_HAS_LIBUNWIND)
-    if (COMPILER_RT_HAS_LIBUNWIND)
-      # If we're omitting default libraries, we might need to manually link in libunwind.
-      # This can affect whether we detect a statically linked libc++ correctly.
-      list(APPEND CMAKE_REQUIRED_LIBRARIES unwind)
-    endif()
-  endif()
 endif ()
 
 if (COMPILER_RT_USE_ATOMIC_LIBRARY)
