diff --git a/toxav/codecs/h264/codec.c b/toxav/codecs/h264/codec.c
index 9799b442..6af320d4 100644
--- a/toxav/codecs/h264/codec.c
+++ b/toxav/codecs/h264/codec.c
@@ -763,9 +763,11 @@ VCSession *vc_new_h264(Logger *log, ToxAV *av, uint32_t friend_number, toxav_vid
     vc->h264_decoder = avcodec_alloc_context3(codec);
 
     if (codec) {
+#if LIBAVCODEC_VERSION_MAJOR < 60
         if (codec->capabilities & AV_CODEC_CAP_TRUNCATED) {
             vc->h264_decoder->flags |= AV_CODEC_FLAG_TRUNCATED; /* we do not send complete frames */
         }
+#endif
 
         if (codec->capabilities & AV_CODEC_FLAG_LOW_DELAY) {
             vc->h264_decoder->flags |= AV_CODEC_FLAG_LOW_DELAY;
@@ -774,7 +776,11 @@ VCSession *vc_new_h264(Logger *log, ToxAV *av, uint32_t friend_number, toxav_vid
         // vc->h264_decoder->flags |= AV_CODEC_FLAG_OUTPUT_CORRUPT;
         vc->h264_decoder->flags |= AV_CODEC_FLAG2_SHOW_ALL;
         // vc->h264_decoder->flags2 |= AV_CODEC_FLAG2_FAST;
-        // vc->h264_decoder->flags |= AV_CODEC_FLAG_TRUNCATED;
+/*
+#if LIBAVCODEC_VERSION_MAJOR < 60
+        vc->h264_decoder->flags |= AV_CODEC_FLAG_TRUNCATED;
+#endif
+*/
         // vc->h264_decoder->flags2 |= AV_CODEC_FLAG2_CHUNKS;
 
         if (H264_DECODER_THREADS > 0) {
