diff -Naur a/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi x/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi
--- a/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi	2023-04-05 09:25:02.000000000 +0000
+++ x/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi	2023-04-07 21:17:30.055634018 +0000
@@ -198,7 +198,7 @@
 		compatible = "pwm-backlight";
 		pwms = <&pwm 0 1000000 0>;
 		brightness-levels = <0 100 500 1000 1500 2000 2500 2800>;
-		default-brightness-level = <7>;
+		default-brightness-level = <3>;
 		enable-gpios = <&gpx3 0 GPIO_ACTIVE_HIGH>;
 		power-supply = <&fet1>;
 		pinctrl-0 = <&pwm0_out>;
