diff -Naur a/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi x/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi
--- a/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi	2023-04-05 09:25:02.000000000 +0000
+++ x/arch/arm/boot/dts/samsung/exynos5250-snow-common.dtsi	2023-04-07 21:17:30.055634018 +0000
@@ -575,7 +575,7 @@
 	status = "okay";
 	non-removable;
 	cap-sdio-irq;
-	keep-power-in-suspend;
+//	keep-power-in-suspend;
 	samsung,dw-mshc-ciu-div = <3>;
 	samsung,dw-mshc-sdr-timing = <2 3>;
 	samsung,dw-mshc-ddr-timing = <1 2>;
